### PR TITLE
feat: status reporting

### LINKED ISSUE #63: Error handling
Things will go wrong and problems like:

 - IPFS hash not found
 - Pinning the hash rejected because the size of the hash exceeds the prepaid hash
 - Pinning will be rejected because it is blacklisted hash
 
In first step we have to mechanism how to communicate these issues and communicate these, hence #62 is needed.

### LINKED ISSUE #142: Basic reporting on pinning process
As first step we should implement basic reporting on pinning process e.g.:

 - pinning started
 - pinning finished
 - pinning errored out

Later on we can improve upon with #77 which might need to add this functionality to js-ipfs though.

### PR DESCRIPTION
**Pull request by AuHau**
Closes #142 and closes #63

### CODE REVIEW

File: src/processor/blockchain-events/agreement.ts [hunk 1]
```diff
@@ -66,7 +70,10 @@ const handlers: HandlersObject<BlockchainAgreementEvents, BlockchainEventProcess
     agreement.isActive = false
     await agreement.save()
 
-    if (options.manager) await options.manager.unpin(agreement.dataReference)
+    if (options.manager) {
+      await options.manager.unpin(agreement.dataReference)
+      channel.broadcast(MessageCodesEnum.I_AGREEMENT_STOPPED, { agreementReference: agreement.agreementReference })
```
Comment (nduchak) [inline src/processor/blockchain-events/agreement.ts:75]: Could we provide a separate class that will handle pinning with jobs, something like ProvideWithJobsManager?
Not sure how much effort it will give us)

Comment (AuHau) [reply src/processor/blockchain-events/agreement.ts:75]: Not sure what you mean 😳 Can you please clarify?

Comment (nduchak) [reply src/processor/blockchain-events/agreement.ts:75]: I mean at some point when we have more logic in our event processing we should think about splitting this part? 
Is the ProviderManager somehow related to Communication?

File: src/communication.ts [hunk 2]
```diff
@@ -25,11 +48,18 @@ class Communication {
...
+  broadcast (code: MessageCodesEnum, payload?: Record<string, any>): void
+  broadcast (code: MessageCodesEnum, payload?: Record<string, any>): void {
     logger.error(`NOT IMPLEMENTED - broadcasting message with code ${code}`, payload)
   }
 }
 
-const instance = Communication.getInstance()
-
-export default instance
+export const channel = Communication.getInstance()
```
Comment (vojtechsimetka) [inline src/communication.ts:65]: Why did you change this from default export?

Comment (AuHau) [reply src/communication.ts:65]: https://github.com/AuHau/tasegir/issues/77

For example it can be subbed for tests, that I forgot to write 😅

Comment (nduchak) [reply src/communication.ts:65]: I found the way of stubbing default import. Check migration PR's test's

File: src/processor/blockchain-events/agreement.ts (hunk 1 above)
Comment (AuHau) [reply src/processor/blockchain-events/agreement.ts:75]: Well this is about Agreements not Pinning Jobs. It is in the `if` because we don't have the channel when precaching.

I agree it is not completely great solution but to me I don't really see major benefit in looking for some other solution atm.
